src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -979,3 +975,27 @@ ColumnType(Oid relationId, char *columnName)
 
 	return columnType;
 }
+
+
+void
+EnsureReplicationSettings(Oid relationId)
+{
+	char replicationModel = (char) ReplicationModel;
+	char *extraHint = " or setting \"citus.replication_model\" to \"statement\"";
+
+	if (relationId != InvalidOid)
+	{
+		replicationModel = TableReplicationModel(relationId);
+		extraHint = "";
+	}
+
+	if (replicationModel == REPLICATION_MODEL_STREAMING && ShardReplicationFactor != 1)
+	{
+		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						errmsg("invalid replication settings"),
```

Comment:
That's a fairly useless primary error message - as so often the detail would be, after slight adaption, a lot better than the current primary message.